server/services/store/sqlstore/board.go

```diff
@@ -763,3 +763,256 @@ func (s *SQLStore) getBoardMemberHistory(db sq.BaseRunner, boardID, userID strin
 
 	return memberHistory, nil
 }
+
+func (s *SQLStore) getBoardsForUserAndTeam(_ sq.BaseRunner, userID, teamID string, includePublicBoards bool) ([]*model.Board, error) {
+	if includePublicBoards {
+		boards, err := s.SearchBoardsForUserInTeam(teamID, "", userID)
```

Comment:
Instead of calling public methods from within the private methods, you should take the `db` argument and use it to call the internal private methods with it. That way, if the `getBoardsForUserAndTeam` was transactional, you'd run all the methods within the context of that transaction.

For this specific case, and given that you'd add the `db` parameter to the signature instead of the `_`, this would be:

```suggestion
		boards, err := s.searchBoardsForUserInTeam(db, teamID, "", userID)
```